source/Calamari.ConsolidateCalamariPackages/ConsolidatedPackageCreator.cs

```diff
@@ -24,39 +24,33 @@ public static void Create(IEnumerable<SourceFile> sourceFiles, string destinatio
 
         private static void WriteUniqueFilesToZip(IEnumerable<SourceFile> sourceFiles, ZipArchive zip)
         {
+            // This acts as a 'Distinct' - there may be multiples of a given file which are binary-identical
+            // therefore we only need to track ONE (aka the first) of these files during packing.
             var uniqueFiles = sourceFiles
-                .GroupBy(sourceFile => new {sourceFile.FullNameInDestinationArchive, sourceFile.Hash})
-                .Select(g => new
-                {
-                    g.Key.FullNameInDestinationArchive,
-                    g.Key.Hash,
-                    g.First().FullNameInSourceArchive,
-                    g.First().ArchivePath
-                });
+                .DistinctBy(sourceFile => sourceFile.EntryNameInConsolidationArchive());
 
             foreach (var groupedBySourceArchive in uniqueFiles.GroupBy(f => f.ArchivePath))
             {
                 using (var sourceZip = ZipFile.OpenRead(groupedBySourceArchive.Key))
                     foreach (var uniqueFile in groupedBySourceArchive)
                     {
-                        var entryName = Path.Combine(uniqueFile.Hash, uniqueFile.FullNameInDestinationArchive);
-                        var entry = zip.CreateEntry(entryName, CompressionLevel.Fastest);
-
+                        var entry = zip.CreateEntry(uniqueFile.EntryNameInConsolidationArchive(), CompressionLevel.Fastest);
```

Comment:
could `CompressionLevel.SmallestSize` be a good way to go? We can afford extra creation time if it makes it a smaller archive